app/main/authorize/keycloak_login_required_decorator.py

```diff
@@ -80,3 +73,46 @@ def _check_if_user_has_access_to_ayr(keycloak_ayr_user_group, decoded_token):
         if keycloak_ayr_user_group in group:
             group_exists = True
     return group_exists
+
+
+def _decode_token(access_token):
+    keycloak_openid = keycloak.KeycloakOpenID(
+        server_url=current_app.config["KEYCLOAK_BASE_URI"],
+        client_id=current_app.config["KEYCLOAK_CLIENT_ID"],
+        realm_name=current_app.config["KEYCLOAK_REALM_NAME"],
+        client_secret_key=current_app.config["KEYCLOAK_CLIENT_SECRET"],
+    )
+
+    decoded_token = keycloak_openid.introspect(access_token)
+
+    return decoded_token
+
+
+def get_user_transferring_body_groups():
```

Comment:
Can we move `get_user_transferring_body_groups` and any share code to a separate `keycloak_user_groups` and just leave the decorator in this keycloak_login_required_decorator.py file? 

Or at least rename this file